pyccel/ast/basic.py

```diff
@@ -21,6 +21,7 @@ class PyccelAstNode:
     _rank      = None
     _dtype     = None
     _precision = None
+    _order     = None
```

Comment:
I don't think the order has been initialised in all the classes which inherit from PyccelAstNode

Reply:
@EmilyBourne For some classes `_rank` and `_order` have no meaning, so there is no reason to give them a value other than the default `None`. Are there classes where we should have initialized them, but we did not?

Reply:
The ones I have found where I think it does need to be set are:
- [ ] PythonTuple
- [ ] Len
- [ ] Array (order is initialised but in Basic not the PyccelAstNode property)
- [ ] Rand
- [ ] Full (order is initialised but in Basic not the PyccelAstNode property)
- [ ] PyccelOperator ( and PyccelDiv )
- [ ] PyccelBooleanOperator
- [ ] PyccelAssociativeParenthesis
- [ ] PyccelUnary
- [ ] DottedVariable
- [ ] FunctionCall
- [ ] IndexedVariable
- [ ] IndexedElement

Reply:
in this PR `_order` is not calculated it's just created for the moment and all the instance that needs it have methods that overrides it.

Reply:
Although I like small pull requests, in this case I think that the calculation of `_order` belongs to this PR, as it would be confusing to keep it separate.

Reply:
adding it will require more changes in different part of the code , we will also need to add more tests to make sure that everything works which is not the intent of this PR

Reply:
OK, I have updated the description of issue #335.